patches/net/minecraft/server/jsonrpc/api/Schema.java.patch

```diff
@@ -0,0 +1,17 @@
+--- a/net/minecraft/server/jsonrpc/api/Schema.java
++++ b/net/minecraft/server/jsonrpc/api/Schema.java
+@@ -95,6 +_,14 @@
+         return schemacomponent;
+     }
+ 
++    public static SchemaComponent registerSchema(net.minecraft.resources.ResourceLocation name, Schema schema) {
++        synchronized (SCHEMA_REGISTRY) {
```

Comment:
You've made this synchronized instead of adding an event. Please add an event (as I've mentioned already, the schema need not be registered immediately after construction.

Reply:
A static method is not any harder to forget, especially given that we design our APIs around events, specifically avoiding any kind of static `register` methods (and we adapt vanilla to this too). This would deviate from that, for no technical reason, causing even more confusion (given the aforementioned API design philosophy). It should be possible to validate all schema components once the server is started (possibly offthread) to make sure they're registered.